### ISSUE TITLE
MapChart JSON Parse Error: Promise Is Not Function 

### ISSUE THREAD

**Issue by ng-druid** [amcharts/amcharts5#1020]
I'm trying to create a MapChart with JSON but run into the below error.

```javascript
 var parser = am5plugins_json.JsonParser.new(root);
  parser.parse({
    refs: {
      data: am5geodata_worldLow
    },
    type: "MapChart",
    settings: {
      panX: "rotateX",
      projection: am5map.geoNaturalEarth1()
    },
    properties: {
      series: [{
        type: "MapPolygonSeries",
        settings: {
          name: "Series"
        },
        properties: {
          geoJSON: "#data"
        }
      }]
    }
  }, {
    parent: root.container
  }).then(function(chart) {
    // This kicks in when config is parsed
    chart.series.getIndex(0).appear(1000);
    chart.appear(1000, 100);
  });
```

> Json.js:132 Uncaught (in promise) TypeError: promise is not a function
    at ParserState.eval (Json.js:132:40)
    at Generator.next (<anonymous>)
    at eval (tslib.es6.mjs:154:69)
    at new Promise (<anonymous>)
    at __awaiter (tslib.es6.mjs:150:10)
    at ParserState.storeClass (Json.js:130:64)
    at ParserState.cacheClass (Json.js:138:50)
    at ParserState.eval (Json.js:182:44)
    at Generator.next (<anonymous>)
    at eval (tslib.es6.mjs:154:69)

It appears that MapPolygonSeries exists using the debugger.

<img width="1439" alt="Screen Shot 2023-07-21 at 7 31 34 PM" src="https://github.com/amcharts/amcharts5/assets/73197190/3dfb254f-ddfe-4a7a-a865-d13aa83fb169">

**Comment by martynasma** [comment]
There are two issues why the map is not working. One is on you, the other is on us :)

`geoJSON` is a setting, not property.

Also, GeoJSON format contains `type` in it, which confuses parser trying to instantiate a class like that, hence the messup.

We'll update the parser to ignore geodata in the next version, and will let you know here.

**Comment by ng-druid** [comment]
@martynasma Thanks for the response. Look forward to the next release with this fixed.

**Comment by martynasma** [comment]
Fixed in [5.4.2](https://github.com/amcharts/amcharts5/releases/tag/5.4.2).

## [5.4.2] - 2023-08-07

### Added
- New method on all axis renderers: `toGlobalPosition(position)`. Converts relative axis position taking into account its current zoom count.
- New setting on `FlowNodes`: `disabledField`. Allows pre-collapsing of nodes using data.

### Fixed
- Fixing the JSON parser to work with `geoJSON`.
- Fixing the JSON parser to work with `stops`.
- Fixing the behavior of `calculateSize` when using `transform: scale`.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
